app/controllers/schools/participants_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class Schools::ParticipantsController < Schools::BaseController
+  skip_after_action :verify_authorized
+  skip_after_action :verify_policy_scoped
+
+  before_action :set_school_and_cohort
+  before_action :set_participant, except: :index
+
+  def index
+    @participants = User.order(:full_name).is_participant.in_school(@school.id)
+  end
+
+  def show
+    @mentor = @participant.early_career_teacher_profile&.mentor
+  end
+
+  def edit_details; end
+
+  def edit_mentor
+    @mentor_form = ParticipantForm.new(
+      mentor_id: @participant.early_career_teacher_profile.mentor&.id,
+    )
+
+    @mentors = @school.mentors.order(:full_name)
+  end
+
+  def update
+    @mentor_form = ParticipantForm.new(participant_form_params)
+
+    # mentor_id can be nil
+    if @mentor_form.mentor_id
+      mentor = @mentor_form.mentor_id != "later" && User.find(@mentor_form.mentor_id)
+
+      if mentor && @school.mentors.exclude?(mentor)
+        raise Pundit::NotAuthorizedError, "No access to this mentor"
+      end
+
+      @participant.early_career_teacher_profile.update!(mentor_profile: mentor ? mentor.mentor_profile : nil)
+
+      flash[:success] = { title: "Success", heading: "The mentor for this participant has been updated" }
+      redirect_to schools_cohort_participant_path(@cohort.start_year, @participant)
+    end
+  end
+
+private
+
+  def set_school_and_cohort
+    @school = current_user.induction_coordinator_profile.schools.first
+    @cohort = Cohort.find_by(start_year: params[:cohort_id])
+  end
+
+  def set_participant
+    @participant = User.is_participant.in_school(@school.id).find(params[:participant_id] || params[:id])
+  end
```

Comment:
This is something that is bogging me for a while. So we have `User` class which might have  anumber of profiles, which is fine. But why do we need to touch `User` record here at all when all we do is operating on their profile? Why instead not query for the profile? I'm fine with this as is, but it's only becuase the proper split of concerns raises a massive amount of questions :D

Reply:
I'm not sure I follow your comment. We need the user, not the profile. Why would we query the profile?

Also, more specifically, "participant" is two profiles, ECT or mentor - my question above is for the more general case.

Reply:
I am not sure it's a silver bullet to operate on profiles rather than on users. In many cases the two are very closely related - for example, creating a new mentor / ect requires both. 

I guess to set a mentor of an ect would be simpler if we gave the induction coordinator the list of mentor profiles to select from (cause that's what is happening under the surface anyway), but I am not convinced the same is true in general.

In this particular case, I think we actively want to handle ects and mentors at the same time - we want to display a list of both ordered by full name, we want to show their details on a very similar page, so it kind of makes sense to me to operate on `User` model. And I think that pattern, of us operating on user model when we want participants is going to be quite prevalent across the app.

Reply:
As mentioned, there are a lot of questions here and a massive potential for a domain modal collapse. For example, we are now allowing updating the email address for a participant/mentor. This makes perfect sense, but in fact, we are updating an underlying user. There's already a discussion about a single user taking multiple roles, so we are at risk for an induction coordinator being able to change an email of another induction coordinator or even, potentially, an admin user? I'd say IC should only be able to edit p[articipant data, and User record, while related to participant, is not a participant data.

I don't think I am able to explain what I am thinking here as it is not 100% clarified in my head neither - I'm just saying that the  data model we're using around User does not translate onto the domain very well and it will, eventually, cause some pain. But it is definitely out of the scope of this PR :D

> creating a new mentor / ect requires both

Absolutely, creating a new mentor is a domain operation and it is absolutely fine for domain operation to create multiple domain models. What about all the cases when we create participant that does not need to log in? Will we create a third type of participant then?

> we want to display a list of both ordered by full name, we want to show their details on a very similar page

This might be flagging that we are just looking at a single Participant record between a school cohort and a profile. In any case, we should not drive the design of the domain by the required UX